protocol/git/README.md

```diff
@@ -0,0 +1,77 @@
+# Git Protocol
+
+A guide for programming within version control.
+
+Maintain a Repo
+---------------
+
+* Use [GitFlow] as a branch management strategy.
+* Avoid including files in source control that are specific to your
+  development machine or process.
+* Delete local and remote feature branches after merging.
+* Perform work in a feature branch.
+* Rebase frequently to incorporate upstream changes.
+* Use a [pull request] for code reviews.
+
+[GitFlow]: https://nvie.com/posts/a-successful-git-branching-model/
+[pull request]: https://help.github.com/articles/using-pull-requests/
+
+Write a Feature
+---------------
+
+Create a local feature branch based off `develop`.
+
+    git checkout develop
+    git pull
+    git checkout -b <branch-name>
```

Comment:
any thoughts on naming strategies? I've found it helpful in the past to have a shared protocol for this. The most effective, in my experience, has been directly naming branches after related tickets in Jira, but more generally relating branches to issue tracking helps keep things clear and feeds back into process well (can't mark a ticket as ready for QA until the ticket's branch is merged to develop, etc)

Reply:
Ah, good question! I'd mentioned GitFlow up top, so that would dictate at least a `feature/whatever` name. Your idea of relating to a ticket is an excellent one. It would be easier if we had a single ticketing system, but we can certainly get something in place here.

Jira: `feature/<ticket id>-human-readable-reminder', e.g. `feature/HG-123-update-readme`. (When using Bitbucket, this format allows the Jira integration to link to the ticket automatically. Don't know if Github has something similar, but it's a decent way to do things regardless, I feel.)

Trello: `feature/<issue ID from link>`, e.g. `feature/51-share-handsome-pov-via-social-media`. (I'm spitballing on this one; better ideas definitely welcome.)

Other: `feature/<short description>`, e.g. `feature/remove-obsolete-links`

Of course, if we can collapse this down to something more concise, that'd be great!

And yes, linking the tickets to the process is a boon for clarity, having a consistent definition of done, etc. 100% in favor.

Reply:
I do agree on always including the ticket ID, then leaving it free form from there. Ideally, our integrations would allow linking the branches to the tickets.

Reply:
Agreed! One of Trello's shortcomings as a ticketing system is that there's not a well-surfaced ticket number; one of several reasons I don't love it for that purpose.